Documentation/ApiOverview/Database/Connection/Index.rst

```diff
@@ -81,6 +81,11 @@ Another way is to inject the :php:`Connection` object directly via
         :language: php
         :caption: EXT:my_extension/Classes/Domain/Repository/MyTableRepository.php
 
+..  warning::
+    Please do not include the `QueryBuilder` using dependency injection, as it
+    will contain various information about the last executed query (stateful).
+    These information can lead to side-effects if further queries will be
+    executed with the `QueryBuilder`.
```

Comment:
```suggestion
    executed with the `QueryBuilder`. Always create a new instance of this class via `GeneralUtility::makeInstance`.
```

Reply:
I believe this is wrong as you should use the function of the connection pool: $queryBuilder = $this->connectionPool
            ->getQueryBuilderForTable('aTable');
            
            I would suggest we also link to that chapter: :ref:`Instantiation <t3coreapi:database-query-builder-instantiation>`

Reply:
Of course, yeah, I forgot a few layers. My mindset was on "use makeInstance to retrieve the connectionpool", but that one is re-usable. Sorry.